### PR TITLE
DOC: replace dec.slow with pytest.mark.slow

### LINKED ISSUE #16184: Deprecate @dec.slow
```bash
~/Projects/numpy$ grep -ri --exclude-dir='build' '@dec.*slow' .
doc/TESTS.rst.txt:  @dec.slow
doc/TESTS.rst.txt:      @dec.slow
numpy/testing/tests/test_decorators.py:        @dec.slow
numpy/testing/_private/decorators.py:      @dec.slow
```

I looked in scipy and `@dec.slow` is only used in one file and seems not to cause tests to end up being not run in `-m fast`.

I'm hoping that I can get approval to delete the mention in 
[doc/TESTS.rst.txt](https://github.com/numpy/numpy/blob/master/doc/TESTS.rst.txt#labeling-tests)

and mark the decorator deprecated (or just delete it) from
[numpy/testing/tests/test_decorators.py](https://github.com/numpy/numpy/blob/master/numpy/testing/tests/test_decorators.py)

### PR DESCRIPTION
**Pull request by rohitsanj**
Resolves #16184

### CODE REVIEW

File: doc/TESTS.rst.txt [hunk 1]
```diff
@@ -136,31 +136,27 @@ modules and develop tests for it.
...
 
   class test_zzz:
-      @dec.slow
+      @pytest.mark.slow
       def test_simple(self):
           assert_(zzz() == 'Hello from zzz')
 
-Available labels are:
+As an alternative to ``pytest.mark.<label>``, there are a number of labels you
+can use:
```
Comment (charris) [inline doc/TESTS.rst.txt:158]: More can be done here. We should not promote any of the nose decorators that can be found in `numpy/testing/tests/test_decorators.py`

Comment (charris) [reply doc/TESTS.rst.txt:158]: There are pytest decorators avaliable for some of these, e.g., `pytest.mark.skipif`. You can grep current numpy to check if numpy uses any of the decorators.

Comment (charris) [reply doc/TESTS.rst.txt:158]: I think all the nose decorators are defined in `numpy/testing/_private/decorators`.

Comment (rohitsanj) [reply doc/TESTS.rst.txt]: Done :+1:

**Comment by charris** [comment]
The error is unrelated, but should be looked at. The warning is issued by `test_public_api.py` when `'numpy.distutils.command.develop'` is imported.

File: doc/TESTS.rst.txt (hunk 1 above)
Comment (rohitsanj) [reply doc/TESTS.rst.txt]: > There are pytest decorators avaliable for some of these, e.g., pytest.mark.skipif. You can grep current numpy to check if numpy uses any of the decorators.

I see that there's another section for `pytest.mark.skipif` - https://github.com/numpy/numpy/blob/master/doc/TESTS.rst.txt#known-failures-skipping-tests

I wil try and grep all such labels and list them. And it seems like [this](https://github.com/numpy/numpy/blob/master/doc/TESTS.rst.txt#parametric-tests) needs to be changed too.

**Review by rgommers — APPROVED** [review]
LGTM, thanks @rohitsanj.

**Comment by mattip** [comment]
Thanks @rohitsanj

**Pull request merged**